ocd-1001.sql

```diff
@@ -0,0 +1,126 @@
+	summary, findings, sites_passed, total_sites, developer_explanation, 
+	resolution, creation_date, last_modified_date, last_modified_user, 
+	deleted)
+	SELECT
+	surveillance_nonconformity_id, surveillance_requirement_id, certification_criterion_id, 
+	nonconformity_type, nonconformity_status_id, date_of_determination, 
+	corrective_action_plan_approval_date, corrective_action_start_date, 
+	corrective_action_must_complete_date, corrective_action_end_date, 
+	summary, findings, sites_passed, total_sites, developer_explanation, 
+	resolution, creation_date, last_modified_date, last_modified_user,
+	deleted
+	FROM surveillance_nonconformity;
+
+INSERT INTO openchpl.surveillance_nonconformity_document 
+	(id, surveillance_nonconformity_id, filename, filetype, filedata, 
+	creation_date, last_modified_date, last_modified_user, deleted)
+	SELECT
+	id, surveillance_nonconformity_id, filename, filetype, filedata, 
+	creation_date, last_modified_date, last_modified_user, deleted
+	FROM surveillance_nonconformity_document;
+
+-- Update sequence numbers
+SELECT pg_catalog.setval(pg_get_serial_sequence('openchpl.surveillance', 'id'), (SELECT MAX(id) FROM openchpl.surveillance)+1);
+SELECT pg_catalog.setval(pg_get_serial_sequence('openchpl.surveillance_requirement', 'id'), (SELECT MAX(id) FROM openchpl.surveillance_requirement)+1);
+SELECT pg_catalog.setval(pg_get_serial_sequence('openchpl.surveillance_nonconformity', 'id'), (SELECT MAX(id) FROM openchpl.surveillance_nonconformity)+1);
+SELECT pg_catalog.setval(pg_get_serial_sequence('openchpl.surveillance_nonconformity_document', 'id'), (SELECT MAX(id) FROM openchpl.surveillance_nonconformity_document)+1);
+
+-- Add function to allow printing text to screen
+CREATE OR REPLACE FUNCTION print_notice(msg text) 
+  RETURNS INTEGER AS 
+$$ 
+DECLARE 
+BEGIN 
+    RAISE NOTICE USING MESSAGE = msg;
+    RETURN null;
+END; 
+$$ 
+LANGUAGE 'plpgsql' IMMUTABLE; 
+
+-- Output certified Product ID for any CAP that is defined as "general " so Jennifer can look at them
+-- Note: "general" would be whenever a CAP does not have an associated certification_criterion; 
+-- in other words, "general" is whenever there is no corrective_action_plan_id in the corrective_action_plan_certification_result table.
+SELECT print_notice('Create temp table called generalCPs with Certified Product IDs for any CAP that is defined as "general"');
+
+DROP TABLE IF EXISTS generalCPs;
+CREATE TEMP TABLE generalCPs AS
+SELECT C.certified_product_id
+FROM openchpl.corrective_action_plan C
+LEFT JOIN openchpl.corrective_action_plan_certification_result CR ON C.corrective_action_plan_id = CR.corrective_action_plan_id
+WHERE C.corrective_action_plan_id IS NOT NULL AND CR.corrective_action_plan_id IS NULL;
+
+SELECT print_notice('Please run the following command to output the table generalCPs to a CSV for Jennifer to review:
+COPY (SELECT * FROM generalCPs) TO ''/generalCPs.copy;''');
```

Comment:
Can this be done automatically? When deploying to PROD we try to avoid requiring any action other that loading the new `vM.N.O.sql` file to reduce downtime / make it closer to automated. As well, does this leave the function in the system? I don't think we want a one-time use function sitting in the data-model any longer than needed. Check `v6.0.0.sql` for an example of a function that was created, ran, then deleted all as part of the deployment